### ISSUE TITLE
DPO Training DataLoader is not shuffled

### ISSUE THREAD

**Issue by kaiwenw** [huggingface/trl#2337]
### System Info

main branch

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

Currently, the training dataloader does not reshuffle data after every epoch. https://github.com/huggingface/trl/blob/main/trl/trainer/dpo_trainer.py#L688

### Expected behavior

Training dataloader should reshuffle.

**Comment by qgallouedec** [comment]
I agree.

Not sure what's the best way to do that though, because it still has to work with the precomputing of ref logprobs. (that's why we initially set `"shuffle": False`). Any idea?

**Comment by sagie2701** [comment]
Hi
Does anyone know how to solve it?
how to set "shuffle": True in the trainer Dataloader

**Comment by qgallouedec** [comment]
Training data is now shuffled in `DPOTrainer`
